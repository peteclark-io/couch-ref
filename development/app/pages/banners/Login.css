@import "../../css/branding.css";

.splash {
   position: absolute;
   top: 0;
   left: 0;
   width: 100%;
   height: 100%;
   background-color: var(--brand-primary)
}

.brand {
   font-family: var(--font-primary);
   color: var(--brand-secondary);
   font-size: 6em;
   text-align: center;
   margin-top: 24vh;
}

.with {
  margin-top: 50px;
  font-family: var(--font-primary);
  color: var(--brand-secondary);
  font-size: 1.4em;
  text-align: center;
}

.auth-providers {
  text-align: center;
}

.auth-providers img {
  width: 50px;
  cursor: pointer;
  margin-left: 10px;
}

.auth-providers img:first-child {
  margin-left: 0px;
}

.facebook {
  padding: 3px;
}
