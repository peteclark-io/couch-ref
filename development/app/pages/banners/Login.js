'use strict';

import React from 'react';
import { connect } from 'react-redux';

import firebase from 'firebase';

import styles from './Login.css';

export const Login = React.createClass({

   propTypes: {
      authenticated: React.PropTypes.bool,
      firebase: React.PropTypes.object
   },

   contextTypes: {
      router: React.PropTypes.object
   },

   componentWillMount: function(){
      this.checkIfAuthenticated();
   },

   componentWillUpdate: function(){
      this.checkIfAuthenticated();
   },

   checkIfAuthenticated: function(){
      if (this.props.authenticated){
         this.context.router.push('/');
      }
   },

   googleLogin: function(){
     var provider = new firebase.auth.GoogleAuthProvider();
     this.props.firebase.auth().signInWithRedirect(provider);
   },

   facebookLogin: function(){
     var provider = new firebase.auth.FacebookAuthProvider();
     this.props.firebase.auth().signInWithRedirect(provider);
   },

   twitterLogin: function(){
     var provider = new firebase.auth.TwitterAuthProvider();
     this.props.firebase.auth().signInWithRedirect(provider);
   },

   render: function() {
       return (
          <div className={styles.splash}>
             <h1 className={styles.brand}>Couch Ref</h1>
             <h2 className={styles.with}>Login With</h2>
             <div className={styles['auth-providers']}>
                <img alt="Login with your Google Account" id="google" src="/images/auth/google.svg" onClick={this.googleLogin} />
                <img alt="Login with Facebook" id="facebook" className={styles.facebook} src="/images/auth/facebook.png" onClick={this.facebookLogin} />
                <img alt="Login with Twitter" id="twitter" className={styles.facebook} src="/images/auth/twitter.svg" onClick={this.twitterLogin} />
             </div>
          </div>
      );
    }
});

const isAuthenticated = (state = {authenticated: false}) => {
   return state.authenticated;
};

const mapStateToProps = (state) => {
   return {
     authenticated: isAuthenticated(state),
     firebase: firebase
   };
};

const LiveLogin = connect(
  mapStateToProps
)(Login);

export default LiveLogin;
