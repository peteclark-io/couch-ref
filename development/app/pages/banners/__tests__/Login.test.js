'use strict';

import React from 'react';
import {Login} from '../Login';
import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer';

const firebase = (checker) => {
   return {
      auth: () => {
         return {signInWithRedirect: (provider) => {checker(provider)}}
      }
   };
}

test('Login snapshot', () => {
   const rendered = renderer.create(
      <Login firebase={firebase((p) => {})} authenticated={false} />
   );

   let tree = rendered.toJSON();
   expect(tree).toMatchSnapshot();
});

it('Can login via google', () => {
   const check = (provider) => {
      expect(provider.providerId).toEqual('google.com');
   };

   const rendered = shallow(
      <Login firebase={firebase(check)} authenticated={false} />
   );

   rendered.find('#google').simulate('click')
});

it('Can login via facebook', () => {
   const check = (provider) => {
      expect(provider.providerId).toEqual('facebook.com');
   };

   const rendered = shallow(
      <Login firebase={firebase(check)} authenticated={false} />
   );

   rendered.find('#facebook').simulate('click')
});

it('Can login via twitter', () => {
   const check = (provider) => {
      expect(provider.providerId).toEqual('twitter.com');
   };

   const rendered = shallow(
      <Login firebase={firebase(check)} authenticated={false} />
   );

   rendered.find('#twitter').simulate('click')
});

it('Redirects if already authenticated', () => {
   var routed = false;
   const router = {
      push: (uri) => {
         expect(uri).toEqual('/');
         routed = true;
      }
   };

   const rendered = mount(
      <Login firebase={firebase((p) => {})} authenticated={true} />,
      {context: {router: router}}
   );

   expect(routed).toBeTruthy();
});
