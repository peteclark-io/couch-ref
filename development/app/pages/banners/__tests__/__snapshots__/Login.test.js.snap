exports[`test Login snapshot 1`] = `
<div
  className="splash">
  <h1
    className="brand">
    Couch Ref
  </h1>
  <h2
    className="with">
    Login With
  </h2>
  <div
    className="auth-providers">
    <img
      alt="Login with your Google Account"
      id="google"
      onClick={[Function]}
      src="/images/auth/google.svg" />
    <img
      alt="Login with Facebook"
      className="facebook"
      id="facebook"
      onClick={[Function]}
      src="/images/auth/facebook.png" />
    <img
      alt="Login with Twitter"
      className="facebook"
      id="twitter"
      onClick={[Function]}
      src="/images/auth/twitter.svg" />
  </div>
</div>
`;
